lib/iris/coords.py

```diff
@@ -2775,6 +2779,366 @@ def xml_element(self, doc):
+    make up a CF-UGRID topology - a constituent of a CF-UGRID mesh.
+
+    See: https://ugrid-conventions.github.io/ugrid-conventions
+
+    """
+
+    def __init__(
+        self,
+        indices,
+        cf_role,
+        long_name=None,
+        var_name=None,
+        attributes=None,
+        start_index=0,
+        element_dim=1,
+    ):
+        """
+        Constructs a single connectivity.
+
+        Args:
+
+        * indices (numpy.ndarray or numpy.ma.core.MaskedArray or dask.array.Array):
+            The index values describing a topological relationship. Constructed
+            of 2 dimensions - the list of elements, and within each element:
+            the indices of the elements it relates to.
+            Use a :class:`numpy.ma.core.MaskedArray` if element lengths vary -
+            mask unused index 'slots' within each element. Use a
+            :class:`dask.array.Array` to keep indices 'lazy'.
+        * cf_role (str):
+            Denotes the topological relationship that this connectivity
+            describes. Made up of this array's elements, and the indexed
+            element within each element. E.g. ``face_node_connectivity``,
+            ``edge_face_connectivity``.
+
+        Kwargs:
+
+        * long_name (str):
+            Descriptive name of the connectivity.
+        * var_name (str):
+            The netCDF variable name for the connectivity.
+        * attributes (dict):
+            A dictionary containing other cf and user-defined attributes.
+        * start_index (int):
+            Either ``0`` or ``1``. Denotes whether :attr:`indices` uses 0- or
+            1-based indexing (allows support for Fortran and legacy NetCDF
+            files).
+        * element_dim (int):
+            Either ``0`` or ``1``. Denotes which dimension of :attr:`indices`
+            refers to the contents of individual elements (allows support
+            for fastest varying index being either first or last). E.g. for
+            ``face_node_connectivity``, where the faces are 3-sided:
+            ``indices.shape[element_dim] = 3``.
+
+        """
+        # Configure the metadata manager.
+        self._metadata_manager = metadata_manager_factory(ConnectivityMetadata)
+
+        self._validate_arg_vs_list("start_index", start_index, [0, 1])
+        # indices array will be 2-dimensional, so must be either 0 or 1.
+        self._validate_arg_vs_list("element_dim", element_dim, [0, 1])
+        # TODO: find a better 'common' location to store valid cf_roles.
+        valid_cf_roles = [
+            "edge_node_connectivity",
+            "face_node_connectivity",
+            "face_edge_connectivity",
+            "face_face_connectivity",
+            "edge_face_connectivity",
+            "boundary_node_connectivity",
+            "volume_node_connectivity",
+            "volume_edge_connectivity",
+            "volume_face_connectivity",
+            "volume_volume_connectivity",
+        ]
```

Comment:
This could be a class attribute, which within the class namespace would give it context e.g., `Connectivity.cf_roles` and would be accessible with or without a `Connectivity` instance.

I suggest `cf_roles` here, rather than `valid_cf_roles` i.e., no need to qualify it, as only those entries in the `Connectivity.cf_roles` are indeed valid.

Just a thought...

Reply:
I originally did this, but it resulted in instances of `Connectivity` having both `my_conn.cf_role` and `my_conn.cf_roles`, which I considered confusing. What do you think?

Reply:
Class attribute called `UGRID_CF_ROLES` should avoid confusion and be a better implementation

Reply:
b9f470a79683d5d0f204ad7183f6749206acfc06